### PR TITLE
DataColumnByRange boilerplate

### PR DESCRIPTION
**Pull request by eserilev**
## Issue Addressed

Boilerplate code for DataColumnsByRange RPC call

### CODE REVIEW

File: beacon_node/network/src/sync/network_context.rs [hunk 1]
```diff
@@ -281,6 +365,46 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
...
+                let blocks_request = Request::BlocksByRange(request);
+
+                // Send both requests. Make sure both can be sent.
+                self.send_network_msg(NetworkMessage::SendRequest {
+                    peer_id,
+                    request: blocks_request,
+                    request_id,
+                })?;
+                self.send_network_msg(NetworkMessage::SendRequest {
+                    peer_id,
```
Comment (dapplion) [inline beacon_node/network/src/sync/network_context.rs]: Is this peer a custodian of the requested columns?

Comment (eserilev) [reply beacon_node/network/src/sync/network_context.rs]: it should be yes, will add a `TODO` here mentioning that we need to implement logic to choose which peers to send the requests to

**Comment by eserilev** [comment]
I'm doing some experimenting on a separate branch: https://github.com/eserilev/lighthouse/commit/d79f0efddcd4e32e69dd5f4ec58d16b6b1e0113a

`send_batch` currently accepts a single peer, and spawns `ByRange` rpc requests to that peer. For data columns we need to: 
- calculate which peers own which columns
- build a list of peers who own the columns we're looking for
- request columns from the list of peers

**Review by dapplion — APPROVED** [review]
Looks good! Note to convert `handle_data_columns_by_range_request` into the same pattern as the rest of handlers that automatically handle errors

**Pull request merged**